zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/JobBasedUserTaskFormTest.java

```diff
@@ -89,7 +169,31 @@ public void shouldResolveAnIncidentIfFormIsDeployed() {
             tuple(incidentCreated.getKey(), IncidentIntent.CREATED),
             tuple(incidentCreated.getKey(), IncidentIntent.RESOLVED));
 
-    assertUserTaskActivation(processInstanceKey);
+    assertUserTaskActivation(processInstanceKey, form.getFormKey());
+  }
+
+  @Test
+  public void shouldRaiseAnIncidentIfFormIsNotDeployedInSameDeploymentForBindingTypeDeployment() {
+    // given
+    deployForm(TEST_FORM_1);
+    final BpmnModelInstance process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .userTask("task")
+            .zeebeFormId(FORM_ID_1)
+            .zeebeFormBindingType(ZeebeBindingType.deployment)
+            .endEvent()
+            .done();
+    final var deployment = ENGINE.deployment().withXmlResource(process).deploy();
+
+    // when
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // then
+    assertFormIncident(
+        processInstanceKey,
+        "Expected to find a form with id '%s' in deployment %s, but not found."
```

Comment:
🔧 Same as with other errors, this could provide some more context on what was happening and a suggestion on how to resolve it.